### ISSUE TITLE
Server crash on undo

### ISSUE THREAD

**Issue by PhanaticD** [IntellectualSites/FastAsyncWorldEdit#514]
# Bug report for FastAsyncWorldEdit 1.15.2
```
[16:26:14] [Paper Watchdog Thread/ERROR]: ------------------------------
[16:26:14] [Paper Watchdog Thread/ERROR]: Current Thread: Server thread
[16:26:14] [Paper Watchdog Thread/ERROR]:       PID: 24 | Suspended: false | Native: false | State: BLOCKED
[16:26:14] [Paper Watchdog Thread/ERROR]:       Thread is waiting on monitor(s):
[16:26:14] [Paper Watchdog Thread/ERROR]:               Locked on:com.boydti.fawe.beta.implementation.chunk.ChunkHolder.call(ChunkHolder.java:729)
[16:26:14] [Paper Watchdog Thread/ERROR]:               Locked on:com.boydti.fawe.beta.implementation.queue.SingleThreadQueueExtent.flush(SingleThreadQueueExtent.java:323)
[16:26:14] [Paper Watchdog Thread/ERROR]:       Stack:
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.boydti.fawe.bukkit.adapter.mc1_15_2.BukkitGetBlocks_1_15_2.call(BukkitGetBlocks_1_15_2.java:298)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.boydti.fawe.beta.implementation.chunk.ChunkHolder.call(ChunkHolder.java:741)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.boydti.fawe.beta.implementation.chunk.ChunkHolder.call(ChunkHolder.java:729)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.boydti.fawe.beta.implementation.queue.SingleThreadQueueExtent.submitUnchecked(SingleThreadQueueExtent.java:171)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.boydti.fawe.beta.implementation.queue.SingleThreadQueueExtent.flush(SingleThreadQueueExtent.java:323)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.boydti.fawe.beta.IQueueExtent.commit(IQueueExtent.java:112)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.sk89q.worldedit.extent.AbstractDelegateExtent.commit(AbstractDelegateExtent.java:274)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.sk89q.worldedit.extent.AbstractDelegateExtent.commit(AbstractDelegateExtent.java:274)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.sk89q.worldedit.extent.AbstractDelegateExtent.commit(AbstractDelegateExtent.java:274)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.sk89q.worldedit.extent.PassthroughExtent.commit(PassthroughExtent.java:213)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.sk89q.worldedit.EditSession.flushQueue(EditSession.java:1013)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.sk89q.worldedit.EditSession.setBlocks(EditSession.java:961)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.sk89q.worldedit.LocalSession.undo(LocalSession.java:564)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.sk89q.worldedit.command.HistoryCommands.undo(HistoryCommands.java:84)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.thevoxelbox.voxelsniper.sniper.Sniper.undo(Sniper.java:308)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.thevoxelbox.voxelsniper.command.executor.UndoExecutor.executeCommand(UndoExecutor.java:36)
[16:26:14] [Paper Watchdog Thread/ERROR]:               com.thevoxelbox.voxelsniper.command.Command.execute(Command.java:46)
[16:26:14] [Paper Watchdog Thread/ERROR]:               org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159)
[16:26:14] [Paper Watchdog Thread/ERROR]:               org.bukkit.craftbukkit.v1_15_R1.CraftServer.dispatchCommand(CraftServer.java:761)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.PlayerConnection.handleCommand(PlayerConnection.java:1861)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.PlayerConnection.a(PlayerConnection.java:1669)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.PacketPlayInChat.a(PacketPlayInChat.java:47)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.PacketPlayInChat.a(PacketPlayInChat.java:5)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:23)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.PlayerConnectionUtils$$Lambda$5265/1280992072.run(Unknown Source)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.TickTask.run(SourceFile:18)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.MinecraftServer.ba(MinecraftServer.java:1105)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.MinecraftServer.executeNext(MinecraftServer.java:1098)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.MinecraftServer.sleepForTick(MinecraftServer.java:1059)
[16:26:14] [Paper Watchdog Thread/ERROR]:               net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:974)
[16:26:14] [Paper Watchdog Thread/ERROR]:               java.lang.Thread.run(Thread.java:748)
[16:26:14] [Paper Watchdog Thread/ERROR]: ------------------------------
```

**[REQUIRED] FastAsyncWorldEdit Configuration Files:** 

log: https://pastebin.com/raw/5qtzrij1
config: https://pastebin.com/raw/d94ghqsE
config-legacy: https://pastebin.com/raw/3DVApknW

**[REQUIRED] FastAsyncWorldEdit Version Number:** 

```
> fawe version
[21:41:12 INFO]: FastAsyncWorldEdit-260 created by Empire92
[21:41:12 INFO]: FAWE version:
 - Date Jun 13, 2020 12:00:00 AM
 - Commit fcc3c34e
 - Build 260
 - Platform Bukkit
[21:41:12 INFO]: Wiki: https://wiki.intellectualsites.com/FastAsyncWorldEdit/index
[21:41:12 INFO]: ------------------ Platforms ------------------
* Bukkit-Official(1.15-260;33c3cb2)

[21:41:12 INFO]: ---------------- Capabilities ----------------
GAME_HOOKS: Bukkit-Official
CONFIGURATION: Bukkit-Official
USER_COMMANDS: Bukkit-Official
PERMISSIONS: Bukkit-Official
WORLDEDIT_CUI: Bukkit-Official
WORLD_EDITING: Bukkit-Official
```
**[REQUIRED] Spigot/Paper Version Number:** 

```
> version
[21:41:28 INFO]: This server is running Paper version git-Paper-350 (MC: 1.15.2) (Implementing API version 1.15.2-R0.1-SNAPSHOT)
[21:41:28 INFO]: Checking version, please wait...
[21:41:28 INFO]: Previous version: git-Paper-653 (MC: 1.13.2)
[21:41:28 INFO]: You are 4 version(s) behind
```
**[REQUIRED] Description of the problem:**

whatever happened crashed the server, looked like undo
**How to replicate**:

not sure
**Plugins being used on the server:**

```
[21:42:09 INFO]: Plugins (79): ArmorStandTools, BannerMaker, BlockAlias*, BlockArmorStandMovement*, BlocksHub, BookExploitFix*, BossShopPro, BungeePortals*, BuycraftX, ccNoSpecTeleport*, Citizens, CleanroomGenerator*, CommandHelper, ConditionalCommands, Creativechunkclean, Cricket*, DeluxeChat, DiceFurniture, DuckTrails*, EasyChairs, EchoPet, Emotes, Essentials, EssentialsSpawn, FastAsyncWorldEdit, FeatherBoard*, FireworkRemover*, FixDoubleLogins*, FreeAutoMessage*, FurnitureLib, FurnitureMaker, GlobalRep*, HeadDatabase*, HideStream*, iControlU, InterestingFish*, JCPhysicsDisabler*, JLib*, LibsDisguises, LightAPI*, LuckPerms, LuckPermsCompat*, Marriage, MCPainter, MiniaturePets*, Multiverse-Core, MVdWPlaceholderAPI*, NametagEdit*, NoCheatPlus, OpenInv, PacketLimiter, PlaceholderAPI, PlotSquared, PlugMan, PremiumVanish, Prism, ProtectionLib*, ProtocolLib, PwnFilter*, RedstoneClockDetector*, SafeArmorStands*, SafeTrade, Saikban-RMI*, Shart, SkullExploitPatch, spark, SuperTrailsPro*, UltimateTroll*, Vault*, ViaBackwards, ViaRewind, ViaRewind-Legacy-Support, ViaVersion, Votifier, VoxelSniper, WDLCompanion*, WorldBorder*, WorldEdit, WorldGuard
```
**Checklist**:

- [x] I included all information required in the sections above
- [] I made sure there are no duplicates of this report [(Use Search)](https://github.com/IntellectualSites/FastAsyncWorldEdit/issues?q=is%3Aissue)
- [] I made sure I am using an up-to-date version of [FastAsyncWorldEdit for 1.15.2](https://ci.athion.net/job/FastAsyncWorldEdit-1.15/)
- [x] I made sure the bug/error is not caused by any other plugin

**Comment by MattBDev** [comment]
@PhanaticD are you still having problems with this?

**Comment by PhanaticD** [comment]
I dont think so, at least I havent noticed any crashes from this in a while

**Comment by MattBDev** [comment]
Since this hasn't reoccured I'll close it.
